lib/services/localService.js

```diff
@@ -27,37 +27,64 @@ module.exports = function (apiPlatformRepository, BPromise,
         var storedFiles = currentState.workspace.files;
-
-        return BPromise.all(localFilePaths.map(function (localFilePath) {
-          // Search file in storedFiles.
-          var existingFile = _.find(storedFiles, 'path', localFilePath);
-
-          // File exists
-          if (existingFile) {
-            // Remove file from stored list if it exists.
-            storedFiles = _.reject(storedFiles, 'path', localFilePath);
-
-            // If content has changed
-            return fileSystemRepository.getFileHash(localFilePath)
-              .then(function (hash) {
-                if (existingFile.hash !== hash) {
-                  result.changed.push(existingFile.path);
-                } else {
-                  result.unchanged.push(existingFile.path);
-                }
-              });
-          } else {
-            result.added.push(localFilePath);
-          }
-        })).then(function () {
-          result.deleted = _.pluck(storedFiles, 'path');
-          return result;
-        });
+        var localDirectoriesPath = currentState.directoriesPath;
+        var storedDirectories = currentState.workspace.directories;
+
+        return BPromise.all([
+            getFileStatus(),
+            getDirectoriesStatus()
+          ])
+          .return(result);
+
+        function getFileStatus() {
+          return BPromise.all(localFilePaths.map(function (localFilePath) {
+            // Search file in storedFiles.
+            var existingFile = _.find(storedFiles, 'path', localFilePath);
+
+            // File exists
+            if (existingFile) {
+              // Remove file from stored list if it exists.
+              storedFiles = _.reject(storedFiles, 'path', localFilePath);
+
+              // If content has changed
+              return fileSystemRepository.getFileHash(localFilePath)
+                .then(function (hash) {
+                  if (existingFile.hash !== hash) {
+                    result.changed.push(existingFile.path);
+                  } else {
+                    result.unchanged.push(existingFile.path);
+                  }
+                });
+            } else {
+              result.added.push(localFilePath);
+            }
+          })).then(function () {
+            result.deleted = _.pluck(storedFiles, 'path');
+          });
+        }
+
+        function getDirectoriesStatus() {
+          localDirectoriesPath.forEach(function (localDirectoryPath) {
+            var existingDirectory = _.find(storedDirectories, 'path',
+              localDirectoryPath);
+            if (existingDirectory) {
+              storedDirectories = _.reject(storedDirectories, 'path',
+              localDirectoryPath);
+            } else {
+              result.addedDirectories.push(localDirectoryPath);
+            }
+          });
+
+          result.deletedDirectories = _.pluck(storedDirectories, 'path');
+
+          return BPromise.resolve();
```

Comment:
There is no need for a promise here.
You can use Promise.all with an array that mixes promises and values, even with all values